src/include/rados/librados.hpp

```diff
@@ -1040,6 +1040,7 @@ namespace librados
     friend class ObjectWriteOperation;  // copy_from needs to see our IoCtxImpl
 
     IoCtxImpl *io_ctx_impl;
+    std::list<std::string> *cached_pool_names;
```

Comment:
Poking with raw pointers is error prone, I know. Storing the _std::list_ just as the member of _IoCtx_ would be much better solution. Unfortunately, it looks like the _const_ variant of _get_pool_name()_ must be able to modify the list too. Maybe could we address that via some kind of smart pointer? On the other hand the librados doesn't seem to use such things currently, so it could be an introduction of unwanted dependency...